pennylane_lightning/core/src/simulators/lightning_kokkos/StateVectorKokkos.hpp

```diff
@@ -862,74 +863,29 @@ class StateVectorKokkos final
-        gates_indices_["DoubleExcitationPlus"]  = GateOperation::DoubleExcitationPlus;
-        gates_indices_["MultiRZ"]               = GateOperation::MultiRZ;
-        gates_indices_["GlobalPhase"]           = GateOperation::GlobalPhase;
-        gates_indices_["CSWAP"]                 = GateOperation::CSWAP;
-        gates_indices_["Toffoli"]               = GateOperation::Toffoli;
-    }
-    /**
-    * @brief Register generator operations in the generators_indices_ attribute:
-    *        an unordered_map mapping strings to GateOperation enumeration keywords.
-    */
+        for (const auto &pair : Pennylane::Gates::Constant::gate_names) {
+            gates_indices_[std::string{pair.second}] = pair.first;
+        }
+    }
+
+    /**
+     * @brief Register generator operations in the generators_indices_
+     * attribute: an unordered_map mapping strings to GeneratorOperation
+     * enumeration keywords.
+     */
     void init_generators_indices_() {
-        generators_indices_["RX"]                    = GeneratorOperation::RX;
-        generators_indices_["RY"]                    = GeneratorOperation::RY;
-        generators_indices_["RZ"]                    = GeneratorOperation::RZ;
-        generators_indices_["ControlledPhaseShift"]  = GeneratorOperation::ControlledPhaseShift;
-        generators_indices_["CRX"]                   = GeneratorOperation::CRX;
-        generators_indices_["CRY"]                   = GeneratorOperation::CRY;
-        generators_indices_["CRZ"]                   = GeneratorOperation::CRZ;
-        generators_indices_["IsingXX"]               = GeneratorOperation::IsingXX;
-        generators_indices_["IsingXY"]               = GeneratorOperation::IsingXY;
-        generators_indices_["IsingYY"]               = GeneratorOperation::IsingYY;
-        generators_indices_["IsingZZ"]               = GeneratorOperation::IsingZZ;
-        generators_indices_["SingleExcitation"]      = GeneratorOperation::SingleExcitation;
-        generators_indices_["SingleExcitationMinus"] = GeneratorOperation::SingleExcitationMinus;
-        generators_indices_["SingleExcitationPlus"]  = GeneratorOperation::SingleExcitationPlus;
-        generators_indices_["DoubleExcitation"]      = GeneratorOperation::DoubleExcitation;
-        generators_indices_["DoubleExcitationMinus"] = GeneratorOperation::DoubleExcitationMinus;
-        generators_indices_["DoubleExcitationPlus"]  = GeneratorOperation::DoubleExcitationPlus;
-        generators_indices_["PhaseShift"]            = GeneratorOperation::PhaseShift;
-        generators_indices_["MultiRZ"]               = GeneratorOperation::MultiRZ;
-        generators_indices_["GlobalPhase"]           = GeneratorOperation::GlobalPhase;
-    }
-    // clang-format on
+        for (const auto &pair : Pennylane::Gates::Constant::generator_names) {
+            generators_indices_[std::string{pair.second.substr(9)}] =
+                pair.first;
+        }
```

Comment:
not sure if I like this over the previous version tbh if these functions will trigger at runtime. Could you rewrite this method to be called at compile time?

Reply:
Should be possible in principle, but I could not find a nice way to write it.